.github/workflows/build.yml

```diff
@@ -0,0 +1,294 @@
+          binary: "dist/${{ env.PKG_NAME }}"
+          deb_depends: "openssl"
+          rpm_depends: "openssl"
+          config_dir: ".release/linux/package/"
+          preinstall: ".release/linux/preinst"
+          postremove: ".release/linux/postrm"
+      - name: Add Linux Package names to env
+        run: |
+          echo "RPM_PACKAGE=$(basename out/*.rpm)" >> $GITHUB_ENV
+          echo "DEB_PACKAGE=$(basename out/*.deb)" >> $GITHUB_ENV
+      - uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.RPM_PACKAGE }}
+          path: out/${{ env.RPM_PACKAGE }}
+      - uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.DEB_PACKAGE }}
+          path: out/${{ env.DEB_PACKAGE }}
+
+  build-darwin:
+    needs: 
+      - get-product-version
+      - set-ld-flags
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        goos: [ darwin ]
+        goarch: [ "amd64", "arm64" ]
+        go: [ "1.17.5" ]
+      fail-fast: true
+    name: Go ${{ matrix.go }} ${{ matrix.goos }} ${{ matrix.goarch }} build
+
+    env:
+      GOPRIVATE: "github.com/hashicorp"
+      GO111MODULE: on
+      LD_FLAGS: ${{ needs.set-ld-flags.outputs.ldflags }}
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go }}
+      - name: Go Build
+        env:
+          GOOS: ${{ matrix.goos }}
+          GOARCH: ${{ matrix.goarch }}
+          CGO_ENABLED: 0
+        run: |
+          mkdir -p dist out
+          unset GOPATH;
+          go build -v -tags "netcgo" -ldflags "${{ env.LD_FLAGS }}" -o dist/ .
+          zip -r -j out/${{ env.PKG_NAME }}_${{ needs.get-product-version.outputs.product-version }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip dist/
+      - uses: actions/upload-artifact@v2
+        with:
+          name: ${{ env.PKG_NAME }}_${{ needs.get-product-version.outputs.product-version }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip
+          path: out/${{ env.PKG_NAME }}_${{ needs.get-product-version.outputs.product-version }}_${{ matrix.goos }}_${{ matrix.goarch }}.zip
+
+  build-docker-light:
+    name: Docker {{ matrix.arch }} light build
+    needs:
+      - get-product-version
+      - build-linux
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: ["arm", "arm64", "386", "amd64"]
+    env:
+      repo: ${{ github.event.repository.name }}
+      version: ${{ needs.get-product-version.outputs.product-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Login to Artifactory
+        uses: docker/login-action@v1
+        with:
+          registry: docker.artifactory.hashicorp.engineering
+          username: ${{ secrets.ARTIFACTORY_RO_USER }}
+          password: ${{ secrets.ARTIFACTORY_RO_TOKEN }}
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+        with:
+          image: docker.artifactory.hashicorp.engineering/prodsec-binfmt:latest
```

Comment:
I believe authenticating to Artifactory shouldn't be needed here, as we now [have sign-off from security to reference a public image](https://github.com/hashicorp/actions-docker-build/pull/8#issuecomment-1035075193) with a verified digest. The QEMU changes, including removing Artifactory login and using public image, are now also [merged into main](https://github.com/hashicorp/actions-docker-build/commit/ddaad7b301cc1b2e7e6201302ded2f3be5472812) in `actions-docker-build`, so we should cut a release of that and then remove these lines along with the GitHub secrets I think (unless they are needed elsewhere?)

@smacfarlane would you like to make a new release of `actions-docker-build` today? As long as the change isn't breaking which I don't think it is, you can add a `v1.2` tag for the GitHub Release itself, and also force-move the `v1` tag to point at `v1.2`. If it's a breaking change we'll need to go to `v2` though.

Reply:
Thanks for the review, Sam! We have a separate Jira ticket to track removing the login to artifactory / setup-qemu steps across projects when there's new version of the docker-build action available. We had to add across multiple projects to prevent docker arm build failures. If the new version is tagged and available before this PR is merged, I'll push up a commit to remove this.

Reply:
Hi @mdeggies we released the update to `actions-docker-build` yesterday, so the `v1` tag you're already using now points to `v1.2` which includes this change, along with the smoke test work done by @jeanneryan.

So, with any luck if you remove this code block things should continue to work. If you hit any issues, please let us know straight away! Thanks :)